### PR TITLE
Proposal for an editor/context menu.

### PR DESCRIPTION
**Pull request by cfehse**
Fixes issue #338.

Introduces the following change(s):
- Configuration for the `editor/context` menu in `package.json`.
- Introduced the to contexts `calva:connected` and `calva:connecting` to enable/disable respectively show/hide menu items according to state.

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Made sure I am directing this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I am changed the default PR base branch, so that it is not `master`. (Sorry for the nagging.)
- [x] Referenced the issue I am fixing/addressing.
     - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
- [x] Created the issue I am fixing/addressing, if it was not present.
- [x] Updated the `[Unrleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.

Ping: @pez, @kstehn

### CODE REVIEW

**Comment by PEZ** [comment]
This is starting to look wonderful! I love that extra nice touch with using enablement.

Please note that @bpringe has remove the commands for pretty printing results: https://github.com/BetterThanTomorrow/calva/pull/344
I think the toggle pprint command should be in the context menu.

**Comment by cfehse** [comment]
@PEZ, @kstehn I merged the latest dev branch and added "append as result" commands. 

**HINT**

I shortened/Simplified the command titles so they fit better into the menu. I also used the vscode convention like "Go to Definition" (all "important" words capitalized) to maintain a similar look in the menu. If we decide to merge these changes, I will check the wiki to change the command titles were needed. The README.md is already changed accordingly.

File: package.json [hunk 1]
```diff
@@ -456,22 +456,22 @@
...
             {
                 "command": "calva.jackIn",
-                "title": "Start a project REPL and connect (aka Jack-In)",
+                "title": "Start a Project REPL and Connect (aka Jack-In)",
                 "category": "Calva"
             },
             {
                 "command": "calva.connect",
-                "title": "Connect to a running REPL server in your project",
+                "title": "Connect to a Running REPL Server in Project",
```
Comment (PEZ) [inline package.json]: Maybe ”the Project”? It's longer, but grammar... :smile:

File: package.json [hunk 2]
```diff
@@ -496,122 +496,122 @@
...
             {
                 "command": "calva.clearInlineResults",
-                "title": "Clear all inline display of evaluation results",
+                "title": "Clear Inline Evaluation Results",
                 "category": "Calva"
             },
             {
                 "command": "calva.evaluateSelection",
-                "title": "Evaluate current form/selection inline and print to output",
+                "title": "Evaluate Current Form Inline",
```
Comment (PEZ) [inline package.json]: I suggest skipping ”Inline”.

File: package.json [hunk 3]
```diff
@@ -496,122 +496,122 @@
...
                 "category": "Calva"
             },
             {
                 "command": "calva.evaluateCurrentTopLevelForm",
-                "title": "Evaluate current top level form inline and print to output (aka defun)",
+                "title": "Evaluate Top Level Form Inline",
```
Comment (PEZ) [inline package.json]: I'd like to keep `defun` in the title. It's what CIDER people will search for.

File: package.json [hunk 4]
```diff
@@ -496,122 +496,122 @@
...
                 "category": "Calva"
             },
             {
                 "command": "calva.evaluateSelectionPrettyPrint",
-                "title": "Evaluate current form/selection, and pretty print to output",
+                "title": "Evaluate Current Form Inline (pretty print)",
```
Comment (PEZ) [inline package.json]: These commands are gone. Or should be, with #344 .

File: package.json [hunk 5]
```diff
@@ -496,122 +496,122 @@
...
             {
                 "command": "calva.evaluateSelectionReplace",
-                "title": "Evaluate current form, and replace it with the result",
+                "title": "Evaluate Current Form and Replace it with Result",
                 "category": "Calva"
             },
             {
                 "command": "calva.evaluateSelectionAsComment",
-                "title": "Evaluate current form, and append a comment with the result",
+                "title": "Evaluate Current Form and Append Comment with Result",
```
Comment (PEZ) [inline package.json]: We could get away with ”Evaluate Current Form to Comment”, I think.

File: package.json [hunk 6]
```diff
@@ -496,122 +496,122 @@
...
             {
                 "command": "calva.loadNamespace",
-                "title": "Load current namespace in REPL window",
+                "title": "Load Current Namespace in REPL Window",
                 "category": "Calva"
             },
             {
                 "command": "calva.setREPLNamespace",
-                "title": "Switch namespace in current window REPL to namespace of this file",
+                "title": "Switch Namespace in Current REPL Window to Current Namespace",
```
Comment (PEZ) [inline package.json]: I think skip ”Current” for the REPL window.

File: package.json [hunk 7]
```diff
@@ -496,122 +496,122 @@
...
                 "category": "Calva"
             },
             {
                 "command": "calva.evalCurrentFormInREPLWindow",
-                "title": "Send current form to REPL window and evaluate it",
+                "title": "Evaluate Current Form in REPL Window",
```
Comment (PEZ) [inline package.json]: Agree!

File: package.json [hunk 8]
```diff
@@ -496,122 +496,122 @@
...
             {
                 "command": "calva.evalCurrentTopLevelFormInREPLWindow",
-                "title": "Send current top level form to REPL window and evaluate it (aka defun)",
+                "title": "Evaluate Top Level Form in REPL Window",
                 "category": "Calva"
             },
             {
                 "command": "calva.switchCljsBuild",
-                "title": "Switch CLJS build",
+                "title": "Switch CLJS Build",
```
Comment (PEZ) [inline package.json]: This switches which CLJS build Calva is connected to. Maybe ”Select CLJS Build Connection”...

File: package.json (hunk 4 above)
Comment (cfehse) [reply package.json]: They are gone. *gg*

**Comment by cfehse** [comment]
@PEZ All suggested changes made.

**Comment by PEZ** [comment]
I think this is ready to get merged and released.

Your call, @kstehn . Merge if you agree. :smile:

**Pull request merged**